presto-main/src/main/java/com/facebook/presto/sql/planner/CanonicalPlanGenerator.java

```diff
@@ -53,11 +52,15 @@
     private final PlanNodeIdAllocator planNodeidAllocator = new PlanNodeIdAllocator();
     private final PlanVariableAllocator variableAllocator = new PlanVariableAllocator();
 
-    public static Optional<CanonicalPlanFragment> generateCanonicalPlan(PlanNode root, PartitioningScheme partitioningScheme)
+    public static Optional<CanonicalPlanFragment> generateCanonicalPlan(PlanNode root, List<VariableReferenceExpression> outputLayout)
```

Comment:
Looking at `visitFilter`, do we need to handle `DynamicFilterPlaceholderFunction` in this case?

Reply:
Good catch. I'm not aware of this. I'll put up a different PR for this. I guess the first step is to return `Optional.empty()` when dynamic filter is enabled :)

Reply:
@highker: I checked how dynamic filter is created in `LocalExecutionPlanner`, and it seems that it is only used in `visitSemiJoin` and `visitJoin` as part of lookup. Current we don't support FRC in those places, so there should not be `DynamicFilterPlaceholderFunction` in the first place.

Handling `DynamicFilterPlaceholderFunction` in `RowExpression` is not easy as that means we would need a specific visitor to detect it. On the other hand, I think a better approach is that whenever we detect dynamic filters as we are creating drivers, we put empty FRC context in `DriverContext`.